ivy_tests/test_ivy/test_frontends/test_jax/test_jax_lax_control_flow_operators.py

```diff
@@ -153,3 +153,51 @@ def _test_body_fn(x, y):
         body_fun=_test_body_fn,
         init_val=x[0],
     )
+
+
+@handle_frontend_test(
+    fn_tree="jax.lax.while_loop",
+    dtype_and_x=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("numeric"),
+        min_value=-1000,
+        max_value=1000,
+        min_num_dims=1,
+        min_dim_size=1,
+    ),
+)
+def test_jax_lax_while_loop(
+    *,
+    dtype_and_x,
+    test_flags,
+    on_device,
+    fn_tree,
+    frontend,
+):
+    def _test_cond_fn(x):
+        def any_negative_real(arr):
+            for elem in arr:
+                if isinstance(elem, (int, float)) and elem < 0:
+                    return True
+                elif isinstance(elem, complex):
+                    return False
+                elif isinstance(elem, (list, tuple)):
+                    if any_negative_real(elem):
+                        return True
+            return False
+
+        return any_negative_real(x)
```

Comment:
```suggestion
    def _test_cond_fn(x):
        for elem in x:
            if isinstance(elem, (int, float)) and elem < 0:
                return True
            elif isinstance(elem, complex):
                return False
            elif isinstance(elem, (list, tuple)):
                if any_negative_real(elem):
                    return True
        return False
```

Minor refactor, do you think this is more organized? `any_negative_real(x)` is used only once, I don't see the point of wrapping it as a function and calling it once. If you do have a specific reason behind this, please let me know :)

Reply:
any_negative_real(x) can be called multiple times, let me explain.

Since the number of dimensions x can have be upto 5(Refer : dtype_and_values function) so to access the values(int, float, complex) stored in x we need to recursively travel x until we reach a leaf node which contains the values(int, float, complex).

Basically, any_negative_real(x) is a function which recurses over a multi-dimensional array until it reaches the leaf node containing the values(int, float or complex). Upon reaching leaf nodes it checks whether any real number is negative. 

Example : Assume we have a 3-d array [[[1, 2], [3, 4]], [[5, 6],[7, 8]]]. To access the element "1" any_negative_real(x) would be called 3 times , removing one square bracket at each iteration. 

first call:         x = [[[1, 2], [3, 4]], [[5, 6],[7, 8]]]
second call :  elem = [[1, 2], [3, 4]]
third call :       elem = [1, 2] 

during third call values 1 , 2 both will be compared to 0.

If you have any questions then please tell me i will be happy to answer :-)

Reply:
Oh sorry, I did not notice the recursive call! My fault, great work :)